Require CLONE_NOOWNERCOPY to be defined for clone on macOS.
https://github.com/ccache/ccache/issues/688
https://github.com/erijo/ccache/commit/a91b4379ef648f8471cfb1f1ea219679b8a02125
--- src/Util.cpp.orig
+++ src/Util.cpp
@@ -72,7 +72,9 @@ extern "C" {
 #ifdef __APPLE__
 #  ifdef HAVE_SYS_CLONEFILE_H
 #    include <sys/clonefile.h>
-#    define FILE_CLONING_SUPPORTED 1
+#    ifdef CLONE_NOOWNERCOPY
+#      define FILE_CLONING_SUPPORTED 1
+#    endif
 #  endif
 #endif
 
